Drop memtrace and pyro-caml-ppx from libs/commons.

memtrace (janestreet fork pin) and pyro-caml-ppx (semgrep fork) are not
available on conda-forge and are only used for optional profiling.
Remove the libraries, the Memtrace.trace_if_requested () call in UCommon and
the [@@@pyro_caml "auto"] attribute in Pcre2_ (which was only meaningful with
pyro-caml-ppx).

--- a/libs/commons/dune
+++ b/libs/commons/dune
@@ -34,7 +34,6 @@
    ; async stuff
    lwt
    alcotest-lwt
-   memtrace
    ; concurrency and parallelism
    parmap
    parallelism
@@ -52,7 +51,6 @@
      ppx_sexp_conv
      ppx_inline_test
      lwt_ppx
-     pyro-caml-ppx
    )
  )
 )
--- a/libs/commons/UCommon.ml
+++ b/libs/commons/UCommon.ml
@@ -63,7 +63,6 @@
 let before_exit = ref []
 
 let main_boilerplate f =
-  Memtrace.trace_if_requested ();
   if not !Sys.interactive then
     exn_to_real_unixexit (fun () ->
         let default_handler signal =
--- a/libs/commons/Pcre2_.ml
+++ b/libs/commons/Pcre2_.ml
@@ -73,10 +73,6 @@
 *)
 let extra_flag = `UTF
 
-(* Auto instrument Pcre_ with profiling since we can spend a lot of time here
-   and FFI calls aren't covered by the memprof profiler *)
-[@@@pyro_caml "auto"]
-
 (*
    Match and depth limits are set explicitly to make semgrep
    fail consistently across platforms (e.g. CI vs. local Mac).
